### DISCUSSION TITLE
Governance and future of the Altair project

### DISCUSSION THREAD

**Discussion by binste** [vega/altair#2834] [General]
Altair has become one of the main data visualization libraries in the Python ecosystem with thousands of users. It has around [~300'000 daily downloads from pypi](https://pypistats.org/packages/altair), countless blog posts have been written about it, there are many university courses which use it, over [1'100 questions asked on StackOverflow](https://stackoverflow.com/questions/tagged/altair), 135 contributors to the `altair` repository, etc. I am a huge fan of Altair myself and greatly appreciate all the work that went into building it up to what it is now, especially by @jakevdp.

Over the last months, @jakevdp did interact with the Altair project much less then he used to, leaving a void not only in terms of actual development time but also knowledge as well as reviewing and merging of pull requests. I want to emphasize that I don't mean this in any negative way, I don't see it as his obligation and without him this project would not exist at all. Thanks to @mattijn and @joelostblom, who became maintainers of the `altair` repo this summer, development continued and picked up a good pace in the last few months with a version 5 in sight.

However, in case @jakevdp does not intend to be more actively involved again in the project, I think a few questions around governance need to be resolved so that new releases of Altair can happen and also to speed up development.

* `altair` repo: As stated above, @mattijn and @joelostblom have maintainer rights which is great (assuming they want to continue). However, as stated in https://github.com/altair-viz/altair/discussions/2599, they do not have the credentials to publish new package releases to pypi (and thereby conda which happens through a feeder repo based on the pypi release)
* Other repos in the `altair-viz` organisation such as `altair_viewer`, `altair_saver`, `altair-viz.github.io`: These packages need to be kept up-to-date with the developments in `altair`. In my understanding, no one except @jakevdp (and @stsievert as organisation member) has maintainer access as well as permissions to release and upload new packages. Furthermore, new repos might need to be set up, for example for a new GitHub page for the Altair 4 documentation once Altair 5 is released

It would be great if we can have a discussion on on how to resolve these points. Naturally, this needs to involve @jakevdp and I really hope you Jake find some time to reply to this and share your thoughts and how you would like to continue with the governance of Altair. Is it an option for you to continue as the main maintainer? Is there a group of people who would be willing to take on more responsibility and take some or all of this over from you? And would you feel comfortable with this? For my part, I definitely want to continue as a contributor to the `altair-viz` family of packages. I'm also open for more if it would be deemed beneficial for the project.

Thanks again to everyone who has contributed to Altair, be it with code, knowledge, or by helping other users! Looking forward to hearing your thoughts.

Some related discussions and issues for further context:
* https://github.com/altair-viz/altair/discussions/2599
* https://github.com/altair-viz/altair/pull/2827
* https://github.com/altair-viz/altair/issues/2793
* https://github.com/altair-viz/altair_viewer/pull/51

**Comment by mattijn** [comment]
Thanks for writing this down @binste! Again, much appreciated. I think everyone involved in the development of Altair is working towards a new release. I think it is good to discuss this further after we have released a release candidate.

**Comment by joelostblom** [comment]
Thank you for taking the time to put this together @binste. I agree that this is an important discussion for us to have and ensure the long term maintainability of Altair, and at the same time we also want to be as supportive as possible of Jake and his future involvement with Altair. As Mattijn mentioned, I think we can start engaging more actively with this discussion once we have received some feedback from the community on a release candidate for Altair and are closer to wanting to make an official release.

**Comment by binste** [comment]
I'm closing this discussion in favor of #3139.
